contracts/cw-names-registry/src/contract.rs

```diff
@@ -0,0 +1,256 @@
+#[cfg(not(feature = "library"))]
+use cosmwasm_std::entry_point;
+use cosmwasm_std::{
+    from_binary, to_binary, Binary, Deps, DepsMut, Empty, Env, MessageInfo, Response, StdResult,
+    Uint128,
+};
+use cw2::set_contract_version;
+use cw20::Cw20ReceiveMsg;
+
+use crate::error::ContractError;
+use crate::msg::{
+    ExecuteMsg, InstantiateMsg, LookUpDaoResponse, LookUpNameResponse, QueryMsg, ReceiveMsg,
+};
+use crate::state::{Config, CONFIG, DAO_TO_NAME, NAME_TO_DAO, RESERVED_NAMES};
+
+const CONTRACT_NAME: &str = "crates.io:cw-name-registry";
+const CONTRACT_VERSION: &str = env!("CARGO_PKG_VERSION");
+
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn instantiate(
+    deps: DepsMut,
+    _env: Env,
+    _info: MessageInfo,
+    msg: InstantiateMsg,
+) -> Result<Response, ContractError> {
+    set_contract_version(deps.storage, CONTRACT_NAME, CONTRACT_VERSION)?;
+
+    let payment_token_address = deps.api.addr_validate(&msg.payment_token_address)?;
```

Comment:
I never considered that we might want to set the payment token (I assumed it would be the native token of the chain), but this is definitely a good call I think.

Reply:
Yeah this uses cw20s, not natives, maybe we could do multiple payment methods?